pkg/apis/kops/cluster.go

```diff
@@ -471,18 +490,54 @@ func (z *ClusterZoneSpec) assignCIDR(c *Cluster) (string, error) {
 	ip4 := cidr.IP.To4()
 	if ip4 != nil {
 		n := binary.BigEndian.Uint32(ip4)
-		n += uint32(index) << uint(32-networkLength)
+		n += uint32(index) << uint(32 - networkLength)
 		subnetIP := make(net.IP, len(ip4))
 		binary.BigEndian.PutUint32(subnetIP, n)
 		subnetCIDR := subnetIP.String() + "/" + strconv.Itoa(networkLength)
+		z.CIDR = subnetCIDR
 		glog.V(2).Infof("Computed CIDR for subnet in zone %q as %q", z.Name, subnetCIDR)
-		return subnetCIDR, nil
+		glog.Infof("Assigned CIDR %s to zone %s", subnetCIDR, z.Name)
+
+		if needsPrivateBlock {
```

Comment:
Are you asking how we are calculating the subnet? Or how we know we need to calculate another block?